hoot-core/hoot-core.pro

```diff
@@ -52,1053 +50,18 @@ include(protobuf.pri)
 
 PRECOMPILED_HEADER = src/main/cpp/hoot/core/HootCoreStable.h
 
+SOURCES += $$files(src/*.cpp, true)
+HEADERS += $$files(src/*.h, true)
+
+# log4cxx
+SOURCES -= src/main/cpp/hoot/core/util/LogLog4Cxx.cpp
+SOURCES -= src/main/cpp/hoot/core/util/LogGeneric.cpp
+HEADERS -= src/main/cpp/hoot/core/util/LogLog4Cxx.h
+HEADERS -= src/main/cpp/hoot/core/util/LogGeneric.h
 log4cxx {
   SOURCES += src/main/cpp/hoot/core/util/LogLog4Cxx.cpp
   HEADERS += src/main/cpp/hoot/core/util/LogLog4Cxx.h
 } else {
   SOURCES += src/main/cpp/hoot/core/util/LogGeneric.cpp
   HEADERS += src/main/cpp/hoot/core/util/LogGeneric.h
 }
-
```

Comment:
Removing all of this from the `pro` file is great.  Instead of removing all of the log stuff and then adding them back in, you could just remove the ones you don't want.  Like this:
```
 log4cxx {
    # remove the generic log files
    SOURCES -= src/main/cpp/hoot/core/util/LogGeneric.cpp
    HEADERS -= src/main/cpp/hoot/core/util/LogGeneric.h
  } else {
    # remove the log4cxx files
    SOURCES -= src/main/cpp/hoot/core/util/LogLog4Cxx.cpp
    HEADERS -= src/main/cpp/hoot/core/util/LogLog4Cxx.h
  }
```
Since they are all already in there, just remove the generic log if `log4cxx` is enabled.